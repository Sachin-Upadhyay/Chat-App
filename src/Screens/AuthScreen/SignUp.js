import React, { useEffect, useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  TouchableOpacity,
  StatusBar,
  Keyboard,
  TouchableWithoutFeedback
} from "react-native";
import { COMPANY_LOGO } from '../../Constants/ImagesPath'
import Style, { PLACEHOLDER_COLOR, PRIMARY_COLOR } from "../../Constants/Style";
import database from '@react-native-firebase/database';
import uuid from 'react-native-uuid';

export default function SignUp(props) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setconfirmPassword] = useState("");
  const [fullName, setFullName] = useState("");
  const [aboutUs,setAboutUS]=useState("I am superstar");

  const registerUser = () => {
    let token = uuid.v4();

    if (email != "" && password != "" && confirmPassword != "" && fullName != "") {
      database()
        .ref('/users/' + token)
        .set({
          email: email,
          password: password,
          fullName: fullName,
          token: token,
          img: "https://randomuser.me/api/portraits/men/41.jpg",
          about: aboutUs,
          lastmsg:""
        })
        .then(() => {
          console.log('Data set.')
          setEmail("")
          setFullName("")
          setPassword("")
          setconfirmPassword("")
          props.navigation.navigate('Login')
        });
    }
    else {
      alert("Fill all details")
    }
  }

  return (

    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
      <View style={Style.login_signUp_container}>
        <StatusBar backgroundColor={PRIMARY_COLOR}></StatusBar>
        <Image style={Style.company_image_in_login_signUp} source={COMPANY_LOGO} />
        <View style={Style.input_container}>
          <TextInput
            style={Style.input_view}
            placeholder="Full Name."
            value={fullName}
            placeholderTextColor={PLACEHOLDER_COLOR}
            onChangeText={(email) => setFullName(email)}
          />
        </View>
        <View style={Style.input_container}>
          <TextInput
            style={Style.input_view}
            placeholder="Email."
            value={email}
            placeholderTextColor={PLACEHOLDER_COLOR}
            onChangeText={(email) => setEmail(email)}
          />
        </View>
        <View style={Style.input_container}>
          <TextInput
            style={Style.input_view}
            placeholder="Password."
            value={password}
            placeholderTextColor={PLACEHOLDER_COLOR}
            secureTextEntry={true}
            onChangeText={(password) => setPassword(password)}
          />
        </View>
        <View style={Style.input_container}>
          <TextInput
            style={Style.input_view}
            placeholder="Confirm Pasword."
            value={confirmPassword}
            placeholderTextColor={PLACEHOLDER_COLOR}
            secureTextEntry={true}
            onChangeText={(password) => setconfirmPassword(password)}
          />
        </View>
        <TouchableOpacity onPress={() => props.navigation.navigate('Login')}>
          <Text style={{
            color: PRIMARY_COLOR,
            fontSize: 16
          }}>
            Login
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => registerUser()} style={[
          Style.button, { marginTop: 20 }]}>
          <Text style={
            Style.buttonText}>
            Sign Up
          </Text>
        </TouchableOpacity>
      </View>
    </TouchableWithoutFeedback>
  );
}