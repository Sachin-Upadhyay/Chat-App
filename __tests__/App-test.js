/**
 * @format
 */

import 'react-native';
import React from 'react';
// import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import SignUp from '../src/Screens/AuthScreen/SignUp';



// jest.mock('react-native-app-intro-slider', () => 'Video');
// jest.mock('@rneui/themed', () => 'Video');
// jest.mock('redux', () => 'Video');
// jest.mock('react-redux', () => 'Video');
// jest.mock('react-native-vector-icons', () => 'Video');
jest.mock('react-native-uuid', () => 'Video');
// jest.mock('react-native-simple-dialogs', () => 'Video');
// jest.mock('react-native-screens', () => 'Video');
// jest.mock('react-native-image-picker', () => 'Video');
// jest.mock('react-native-default-preference', () => 'Video');
// jest.mock('@rneui/base', () => 'Video');
// jest.mock('@reduxjs/toolkit', () => 'Video');
// jest.mock('@react-navigation/native-stack', () => 'Video');
// jest.mock('@react-navigation/native', () => 'Video');
// jest.mock('@react-navigation/bottom-tabs', () => 'Video');
jest.mock('@react-native-firebase/database', () => 'Video');
// jest.mock('@react-native-firebase/app', () => 'Video');
// jest.mock('@react-native-async-storage/async-storage', () => 'Vid22eo');
// jest.mock('react-native-vector-icons/FontAwesome', () => 'Vid22eo');
// jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Vid22eo');

it('renders correctly', () => {
  const wrapper= renderer.create(<SignUp />);
  // console.log(wrapper)
  expect(wrapper).toBeDefined();

    // expect(wrapper).toMatchSnapshot();
});

it('function Testing',()=>{
  const wrapper=renderer.create(
    <App/>
  ).getInstance()
  expect(wrapper.change(2)).toEqual(20)
});

it('state Testing',()=>{
  const wrapper=renderer.create(
    <App/>
  ).getInstance()
  wrapper.change(2)
  expect(wrapper.state.data).toEqual(20)
})

let findElement=function(tree,element){
  let result=undefined;
  for(node in tree.children){
    if(tree.children[node].props.testID==element){
      result=true
    }
  }
  return result
}

it('find element',()=>{
  const tree=renderer.create(
    <App/>
  ).toJSON()
  expect(findElement(tree,'username')).toBeDefined();
})
