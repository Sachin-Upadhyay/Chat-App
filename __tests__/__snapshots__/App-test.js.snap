// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2c3e50",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    MyClass
  </Text>
</View>
`;
